### DISCUSSION TITLE
Reverse the import/export logic for the datasources and dashboards

### DISCUSSION THREAD

**Discussion by gtarnaras** [esnet/gdg#45] [General]
Hey guys, thank you very much for this tool. It is really useful and I was very happy to find it (and not having to develop something from scratch!). It works just fine so far but I had a little problem understanding that import/export means import/export to **local file system** and not Grafana. 

I first tried to export the dashboards from grafana to my local filesystem without reading the docs and I could not make it work (and I was not sure what I was doing wrong). Well, after 10 mins I figured out that I was too lazy to read the docs word by word (since the instructions mention this clearly). However, in my mind the import/export target was Grafana and not my local filesystem. As such I though to mention this here and suggest to change this logic. 
 
I believe it is going to be more clear to reverse the logic and have grafana be the target for the import/export operations. 
What I am suggesting is the following logical change:

import/export <object> to/from grafana 
instead of  
import/export <object> from/to grafana 

and the docs would be updated as below:

Current command:
` ./bin/gdg dash import -- Import all dashboards from grafana to local file system`

Suggested change:
` ./bin/gdg dash import -- Import all dashboards from local file system to grafana`

**Comment by safaci2000** [comment]
I was going back and forth on the import/export terminology.  GDG is trying to recreate what [wizzy](https://grafana-wizzy.com/remote-commands/dashboard-commands/) used to be.  I ended up sticking to their terminology.  

It also makes sense to me if I think as grafana the entity we're operating against and the command is the action being performed.

gdg dash clear (clears items from grafana) 
gdg dash import (imports data from grafana)
gdg dash export (exports data to grafana).

I'd rather not flip this behavior as it does get confusing where import means something in 1.6.0 and the opposite in 1.7.0.  


I added two Emojis to this request if we get an overwhelming thumbs up, might revisit but seems like a minor thing isn't it? 

I can add more logging and docs if it makes clearer but I'd rather not flip the meaning.

**Reply by gtarnaras** [reply]
Thanks @safaci2000 . Haven't used wizzy but since cdg is sort of "mimicking" its logic it does make sense to leave this as is. Maybe giving a bit more info in the _Available Commands_ section would make it more clear but again, it is a really minor thing and it is pretty clear in the docs already ( classic RTFM issue :stuck_out_tongue: ).

e.g. 
```
Available Commands:
  clear       delete all monitored dashboards
  export      Exports all dashboard from local filesystem (matching folder filter) to Grafana
  import      Import all dashboards from grafana to local file system
  list        List all dashboards
```

**Reply by L4R5** [reply]
I just deleted all dashboards by "exporting" them :) (I had a backup)

I agree to @gtarnaras, a simple enahancement in the sub command description would help a lot.

**Reply by safaci2000** [reply]
What would you suggest @L4R5.  It already describes the behavior in the command as seen above.  I'm happy to add a longer description.  We can add a warning before all the exports though that'll take some time to update all the various exports for all entities.

**Reply by L4R5** [reply]
I would like the longer descriptions as suggested by @gtarnaras 

I installed the latest version using the deb package and it looks like this with the current short descriptions.
```
~ # gdg version
Build Date:
Git Commit:
Version: 0.4.3
Go Version: go1.20.3
OS / Arch: linux amd64

~ # gdg dashboard
Manage Grafana Dashboards.

Usage:
  gdg dashboards [command]

Aliases:
  dashboards, dash, dashboard

Available Commands:
  clear       delete all monitored dashboards
  export      export all dashboards
  import      Import all dashboards
  list        List all dashboards

Flags:
  -d, --dashboard string   filter by dashboard slug
  -f, --folder string      Filter by Folder Name (Quotes in names not supported)
  -h, --help               help for dashboards
  -t, --tags strings       Filter by Tags (does not apply on export)

Global Flags:
  -c, --config string   Configuration Override

Use "gdg dashboards [command] --help" for more information about a command.
```
